heudiconv/convert.py

```diff
@@ -851,6 +851,57 @@ def nipype_convert(
     return eg, prov_file
 
 
+def filter_partial_volumes(
+    nii_files: list[str],
+    bids_files: list[str],
+    bids_metas: list[dict[str, Any]],
+):
+    """filter interrupted 4D scans volumes with missing slices on XA: see dcm2niix #742
+
+    Parameters
+    ----------
+    nii_files : list[str]
+        converted nifti filepaths
+    bids_files: list[str]
+        converted BIDS json filepaths
+    bids_metas : list[dict[str, Any]]
+        list of metadata dict loaded from BIDS json files
+
+    Returns
+    -------
+    nii_files
+        filtered niftis
+    bids_files
+        filtered BIDS jsons
+    bids_metas
+        filtered BIDS metadata
+
+    """
+    partial_volumes = [not metadata.get("RawImage", True) for metadata in bids_metas]
+    no_partial_volumes = not any(partial_volumes) or all(partial_volumes)
```

Comment:
sorry, my had hurts a little trying to do this `bool` and need for that `or`. Wouldn't it make it easier to talk in terms of 

```suggestion
    has_partial_volumes = any(partial_volumes)
```
? (and then flip `if` below)  or it wouldn't be sufficient?

Reply:
I am not sure if "RawImage":False is also present for other derived data, so I tried to narrow down the case where a series produced niftis with both False and not present (or True if ever). 
There might still be derived data that we want to convert.